libgalois/include/katana/GraphTopology.h

```diff
@@ -1428,6 +1430,79 @@ class KATANA_EXPORT PGViewCache {
 KATANA_EXPORT GraphTopology CreateUniformRandomTopology(
     const size_t num_nodes, const size_t edges_per_node) noexcept;
 
+template <bool IS_SYMMETRIC = false>
+class KATANA_EXPORT TopologyBuilderImpl : public GraphTopologyTypes {
+  using AdjVec = std::vector<Node>;
+
+  std::vector<AdjVec> all_nodes_adj_;
+
+  bool IsValidNode(Node id) const noexcept {
+    return id < all_nodes_adj_.size();
+  }
+
+  void AddEdgeImpl(Node src, Node dst) noexcept {
+    KATANA_LOG_DEBUG_ASSERT(IsValidNode(src));
+    auto& adj_list = all_nodes_adj_[src];
+
+    [[maybe_unused]] bool not_found =
+        (std::find(adj_list.begin(), adj_list.end(), dst) == adj_list.end());
+    KATANA_LOG_DEBUG_ASSERT(not_found);
+
+    adj_list.emplace_back(dst);
+  }
+
+public:
+  void AddNodes(size_t num) noexcept {
+    all_nodes_adj_.resize(all_nodes_adj_.size() + num);
+  }
+
+  void AddEdge(Node src, Node dst) noexcept {
+    AddEdgeImpl(src, dst);
+    if (IS_SYMMETRIC) {
+      AddEdgeImpl(dst, src);
+    }
+  }
+
+  size_t degree(Node src) const noexcept {
```

Comment:
I don't know enough about our coding style guide yet, but my initial impression is that we shouldn't mix Pascal case and snake case (e.g. `AddEdge` vs `num_nodes`).

Reply:
Yeah, I wish it was all one style. We have decided to follow the silly Google Style Guide (parts of which infuriate me). Some of our older code follows `addEdges()` and has not been updated.